common/data/src/main/java/org/thingsboard/server/common/data/query/AssetTypeFilter.java

```diff
@@ -15,18 +15,41 @@
  */
 package org.thingsboard.server.common.data.query;
 
-import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.springframework.util.CollectionUtils;
 
-@Data
+import java.util.Collections;
+import java.util.List;
+
+@ToString
+@EqualsAndHashCode
+@Setter
 public class AssetTypeFilter implements EntityFilter {
 
+    /**
+     * Replaced by {@link AssetTypeFilter#getAssetTypes()} instead.
+     */
+    @Deprecated(since = "3.5", forRemoval = true)
+    private String assetType;
+
+    private List<String> assetTypes;
+
+    public List<String> getAssetTypes() {
+        if (assetType != null) {
```

Comment:
This check is wrong from a forward-compatibility point of view. You should do something like this:

```
return CollectionUtils.isNotEmpty(assetTypes) ? assetTypes : Collections.singletonList(assetType);
```

My logic is the following:
if new variable assetTypes is set, we should always use it. If not, we should use old variable. Otherwise, if someone update the TB version and uses new UI to add multiple asset types, they will not be able to use it if ```assetType``` is stored in the json file of the dashboard alias.